web/src/compose_setup.js

```diff
@@ -179,9 +181,16 @@ export function initialize() {
                     // user-triggered edit to that field.
                     $input.trigger("input");
 
-                    // TODO: Probably this should also renarrow to the
-                    // new topic, if we were currently viewing the old
-                    // topic, just as if a message edit had occurred.
+                    // renarrow to the new topic if currently viewing the old topic
+                    const terms = hash_util.parse_narrow(window.location.hash.split(/\//));
+                    if (terms.length < 2) {
+                        return;
+                    }
+                    const stream_name = sub_store.maybe_get_stream_name(stream_id);
```

Comment:
You don't need to parse the URL; functions of the `Filter` object returned by `narrow_state.current_filter()` to analyze our current view.

Reply:
Sure, I've switched to using those methods.